src/DropdownBase.js

```diff
@@ -0,0 +1,307 @@
+import React, { cloneElement } from 'react';
+import keycode from 'keycode';
+import DropdownToggle from './DropdownToggle';
+import DropdownMenu from './DropdownMenu';
+import MenuItem from './MenuItem';
+import CustomPropTypes from './utils/CustomPropTypes';
+import createChainedFunction from './utils/createChainedFunction';
+
+import find from 'lodash/collection/find';
+
+const TOGGLE_REF = 'toggle-btn';
+
+export default class DropdownBase extends React.Component {
+  constructor(props, Toggle=DropdownToggle) {
```

Comment:
I believe the 2nd param to `React.Component#constructor` is supposed to be `context`.

Could we just use `this.constructor.Toggle` below? Or set `Toggle` on the prototype?